### PR TITLE
MGDOBR-1179 Performance tests: Store benchmark report as a file

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-1179](https://issues.redhat.com/browse/MGDOBR-1179)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by sutaakar** [comment]
This benchmark report is a HTML file with execution overview.
It is intended for local test result examination without Horreum interactions.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HyperfoilResource.java [hunk 1]
```diff
@@ -66,6 +68,17 @@ public static boolean containsFailedRunPhase(String idRun) {
         return run.getPhases().stream().anyMatch(Phase::getFailed);
     }
 
+    public static void storeBenchmarkReport(String perfTestName, String runId) {
+        try {
+            String generateReport = validation.generateReport(validation.getRun(runId));
+            Path destination = Paths.get(BENCHMARK_REPORT_DESTINATION_FOLDER_PATH, perfTestName + "-" + LocalDateTime.now() + ".html");
```
Comment (afalhambra) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HyperfoilResource.java:74]: maybe it's useful to add the `runId` param as part of the name of the generated html file, wdyt?

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HyperfoilResource.java [hunk 2]
```diff
@@ -66,6 +68,17 @@ public static boolean containsFailedRunPhase(String idRun) {
...
+            Files.createDirectories(destination.getParent());
+            Files.writeString(destination, generateReport);
+        } catch (IOException e) {
+            throw new RuntimeException("error while generating benchmark report", e);
+        }
```
Comment (afalhambra) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HyperfoilResource.java:79]: do we really need to stop the execution if unable to store this file locally? IMHO if the test is not executed locally, I think we should skip this exception, and just log a warning - wdyt?

**Comment by sutaakar** [comment]
@afalhambra Suggestions implemented

**Review by afalhambra — APPROVED** [review]
Great work @sutaakar as always!

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**